django/cantusdb_project/main_app/urls.py

```diff
@@ -95,6 +101,45 @@
         views.change_password,
         name="change-password",
     ),
+    # password reset views
+    path(
+        # here, user can initiate a request to send a password reset email
+        "reset-password/",
+        PasswordResetView.as_view(template_name="registration/password_reset.html"),
+        name="password_reset",
+    ),
+    path(
+        # we display this page once the password reset email has been sent
+        "reset-password-sent/",
+        PasswordResetDoneView.as_view(
+            template_name="registration/password_reset_sent.html"
+        ),
+        name="password_reset_done",
+    ),
+    path(
+        # here, the user can specify their new password
+        "reset/<uidb64>/<token>",
+        # Template filename below includes an intentional trailing underscore.
+        # Without this extra underscore, django will find a django built-in template when
+        # it looks for registration/password_reset_confirm.html, rather than finding our
+        # own custom template
+        PasswordResetConfirmView.as_view(
+            template_name="registration/password_reset_confirm_.html"
```

Comment:
Good catch! I don't really like the trailing underscore though. Can this be renamed to something like reset_password_confirm.html to avoid the built in django template but have a nicer looking filename? You could similarly change the other templates to start with reset_password to maintain continuity.